In [0]:
import numpy as np
import pandas as pd

import scipy.stats as stats

In [0]:
pesos = [100, 150, 150, 200, 250, 300, 325, 400, 415, 500, 600, 1000]

In [7]:
#Media
pesos_media = np.mean(pesos)
pesos_media

365.8333333333333

In [8]:
#Mediana
np.median(pesos)

312.5

In [9]:
#Moda
stats.mode(pesos)

ModeResult(mode=array([150]), count=array([2]))

In [10]:
#Rango
np.max(pesos) - np.min(pesos)
#Diferencia entre el mayor y el menor

900

In [11]:
#Cuartiles e IQR
stats.mstats.mquantiles(pesos)

array([172.5 , 312.5 , 461.75])

In [12]:
#La función pandas.qcut convierte una lista de N cuantiles. Si le pasamos 4, tenemos los cuartiles. 
#qcut devuelve una serie categórica nueva con los cuantiles como categorías.
pesos_cuartiles = pd.qcut(pesos, 4)
pesos_cuartiles.categories

IntervalIndex([(99.999, 187.5], (187.5, 312.5], (312.5, 436.25], (436.25, 1000.0]]
              closed='right',
              dtype='interval[float64]')

Aquí se ve que los intervalos que separan los pesos en 4 bloques son:<br>
Categories(4, interval[float64]): [(99.999, 187.5] < (187.5, 312.5] < (312.5, 436.25] < (436.25, 1000]]
<br>
Es importante notar que hay diversas formas de obtener los cuartiles, de aquí la diferencia con el método calculado a mano.

In [13]:
#Desviación standard
pesos_std = np.std(pesos)
pesos_std

239.6771555423856

Coeficiente de simetría <br>
El coeficiente de simetría (skewness) se define como el tercer momento dividido por la desviación standard al cubo. 

In [0]:
pow3 = lambda x: x*x*x

In [0]:
tercer_momento = lambda x: pow3(x - pesos_media)

In [17]:
simetria_pesos = sum(map(tercer_momento, pesos)) / (12*pow3(pesos_std))
simetria_pesos
#Map funciona como el apply de R para evitar hacer un for

1.3623858394083475

In [18]:
#Lo mismo, pero precargado
stats.skew(pesos)

1.3623858394083481

In [23]:
pow4 = lambda x: x*x*x*x
cuarto_momento = lambda x: pow4(x - pesos_media)
curtosis_pesos = sum(map(cuarto_momento, pesos)) / (12*pow4(pesos_std)) - 3
curtosis_pesos

1.4285722765161841

In [24]:
stats.kurtosis(pesos)

1.4285722765161841